core/src/main/java/org/apache/iceberg/BaseIncrementalAppendScan.java

```diff
@@ -83,7 +83,8 @@ private CloseableIterable<FileScanTask> appendFilesFromSnapshots(List<Snapshot>
                     snapshotIds.contains(manifestEntry.snapshotId())
                         && manifestEntry.status() == ManifestEntry.Status.ADDED)
             .specsById(table().specs())
-            .ignoreDeleted();
+            .ignoreDeleted()
+            .columnsToKeepStats(context().columnsToKeepStats());
```

Comment:
I think you have defined `columnsToKeepStats()` method, which you can call directly now.
See `BaseScan`.